pkg/controller/registry/resolver/cache.go

```diff
@@ -306,3 +337,60 @@ func (n *NamespacedOperatorCache) GetPackageVersionFromAllCatalogs(pkg string, v
 	}
 	return result, nil
 }
+
+func (n *NamespacedOperatorCache) GetRequiredAPIFromAllCatalogs(requiredAPI registry.APIKey) ([]Operator, error) {
+	var result []Operator
+	for _, s := range n.snapshots {
+		result = append(result, s.Find(func(o *Operator) bool {
+			providedAPIs := o.ProvidedAPIs()
+			if _, providesRequirement := providedAPIs[requiredAPI]; providesRequirement {
+				return true
+			}
+			return false
+		})...)
+	}
+	if len(result) == 0 {
+		return nil, fmt.Errorf("operator with requiredAPI %s not found in any catalog", requiredAPI)
+	}
+	return result, nil
+}
+
+func (n *NamespacedOperatorCache) GetCsvFromAllCatalogsWithFilter(csvName string, filter installableFilter) ([]Operator, error) {
+	var result []Operator
+	for _, s := range n.snapshots {
+		result = append(result, s.Find(func(o *Operator) bool {
+			candidate := true
+			if filter.channel != "" && o.Bundle().GetChannelName() != filter.channel {
+				candidate = false
+			}
+			if !filter.catalog.IsEmpty() && !filter.catalog.IsEqual(o.SourceInfo().Catalog) {
+				candidate = false
+			}
+			return candidate && o.name == csvName
+		})...)
+	}
+	if len(result) == 0 {
+		return nil, fmt.Errorf("operator with csvName %s not found in any catalog", csvName)
+	}
+	return result, nil
+}
+
+func (n *NamespacedOperatorCache) GetPackageChannelFromCatalog(pkg, channel string, catalog CatalogKey) ([]Operator, error) {
+	var result []Operator
+	s, ok := n.snapshots[catalog]
+	if !ok {
+		return nil, fmt.Errorf("catalog %s not found", catalog)
+	}
+	result = s.Find(func(o *Operator) bool {
+		return o.Bundle().GetChannelName() == channel && o.Package() == pkg
+	})
+	if len(result) == 0 {
+		return nil, fmt.Errorf("operator %s not found in channel %s in catalog %s", pkg, channel, catalog)
+	}
+
+	return result, nil
+}
+
+func (n *NamespacedOperatorCache) GetCacheCatalogSize() int {
```

Comment:
The name of this method is a little ambiguous. It sounds like it could be referring to the number of entries within a catalog, but it's returning the number of catalogs. Maybe `GetCatalogCount`?

Reply:
I think this is something we can address at a later PR.